airbyte-webapp/src/components/GeographyDropdown/GeographyDropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+import { components, OptionProps, MenuListProps } from "react-select";
+
+import "/node_modules/flag-icons/css/flag-icons.min.css";
+
+import { DropDown } from "components/ui/DropDown";
+import { DropdownProps } from "components/ui/DropDown";
+
+import styles from "./GeographyDropdown.module.scss";
+
+const GeographyOption: React.FC<OptionProps> = (props) => {
+  const { formatMessage } = useIntl();
+  return (
+    <components.Option className={styles.option} {...props}>
+      <div className={styles.flag}>
+        <span className={`fi fi-${props.label === "auto" ? "us" : props.label}`} />
+      </div>
+      <span className={styles.label}>{formatMessage({ id: `geography.${props.label}` })}</span>
+    </components.Option>
+  );
+};
+
+const MenuList = (props: MenuListProps) => {
+  return (
+    <components.MenuList {...props}>
+      {props.children}
+      <div className={styles.menuList}>
+        <FontAwesomeIcon icon={faPlus} />
+        <span>Request a new geography</span>
+      </div>
+    </components.MenuList>
+  );
+};
+
+interface Props<T = unknown> extends DropdownProps {
+  options: T[];
+}
+
+export const GeographyDropdown: React.FC<Props> = ({ options }) => {
```

Comment:
We can improve the typing here - `react-select` needs to know what shape an `option` has so that it can correctly type all sorts of other things for us.

```suggestion
interface GeographySelectOption {
  label: Geography;
  value: Geography;
}

export const GeographyDropdown: React.FC<DropdownProps<GeographySelectOption>> = ({ options }) => {
```

Here I added a `GeographySelectOption` which describes our custom "shape" of an option. We can pass this in to `DropdownProps` so that the underlying `react-select` components and types know what kind of options we are passing in.